Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,43 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(
+        Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        TensorDesc outputCountTensor(DML_TENSOR_DATA_TYPE_UINT32, {1,1,1,1}, builder->GetTensorPolicy()); // Same as input
+
+        uint32_t effectiveRank = -1;
+        auto inputTensorSizes = inputTensor.sizes;
```

Comment:
`inputTensorSizes` appears read-only. So avoid copy of `std::vector` and just use `auto&` (or use `inputTensor.sizes` directly, since we're only saving a single character in name length).

Reply:
Using const auto& instead